### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by amyblais**
I recently updated the Code Review and Contribution Checklist dev docs via this PR https://github.com/mattermost/mattermost-developer-documentation/pull/778/files to add a step to list QA test steps in the PR description (if applicable), and thought that we could also update the PR template to say the same.

More context here: https://github.com/mattermost/mattermost-developer-documentation/pull/778#issue-596759155.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 #### Summary
 <!--
-A description of what this pull request does.
+A description of what this pull request does, as well as QA test steps (if applicable).
```
Comment (hanzei) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: Should we mention that the JIRA ticket is the primary place to put QA test steps?

**Review by hanzei — APPROVED** [review]
LGTM, one none blocking question

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (amyblais) [reply .github/PULL_REQUEST_TEMPLATE.md]: I added a note. QA doesn't mind having it in the PR description since most of the testing is done on PRs.

**Review by mkraft — APPROVED** [review]

**Pull request merged**